tools/amd_build/patches/a_torch_csrc_generic_StorageSharing.cpp.patch

```diff
@@ -0,0 +1,28 @@
+diff --git a/torch/csrc/generic/StorageSharing.cpp b/torch/csrc/generic/StorageSharing.cpp
+index 5de9e7e4d..333bc803a 100644
+--- a/torch/csrc/generic/StorageSharing.cpp
++++ b/torch/csrc/generic/StorageSharing.cpp
+@@ -1,8 +1,14 @@
+-#ifdef WITH_CUDA
++#ifdef WITH_ROCM
+ #include <cuda.h>
+ #include <cuda_runtime.h>
+ #endif
+ 
++#if defined(__HIP_PLATFORM_HCC__)
++  #undef PyBytes_AS_STRING(op)
++  #undef PyBytes_GET_SIZE(op)
++  #define PyBytes_AS_STRING(op) (((PyBytesObject *)(op))->ob_sval)
++  #define PyBytes_GET_SIZE(op)  Py_SIZE(op)
++#endif
```

Comment:
But why? Those shouldn't matter for HIP. This file should be still built using the regular C++ compiler